### ISSUE TITLE
kubevirt-api TLS handshake error

### ISSUE THREAD

**Issue by terinjokes** [kubevirt/kubevirt#1832]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

KubeVirt components started up successfully, but no VMs are created or scheduled. Lots of TLS failures in the logs.

```console
$ kubectl logs virt-controller-679d8dc55b-smx9s
{"component":"","level":"info","msg":"kubevirt-config ConfigMap does not exist. Using defaults.","pos":"feature-gates.go:66","timestamp":"2018-12-10T21:18:13.033713Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.033771Z"}
{"component":"virt-controller","level":"info","msg":"DataVolume integration disabled","pos":"application.go:176","timestamp":"2018-12-10T21:18:13.035802Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.035848Z"}
{"component":"virt-controller","level":"info","msg":"attempting to acquire leader lease  kubevirt/virt-controller...","pos":"leaderelection.go:185","timestamp":"2018-12-10T21:18:13.580470Z"}
{"action":"listening","component":"virt-controller","interface":"0.0.0.0","level":"info","port":8443,"pos":"application.go:202","service":"http","timestamp":"2018-12-10T21:18:13.580450Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:27.638002Z"}
{"component":"virt-controller","level":"info","msg":"successfully acquired lease kubevirt/virt-controller","pos":"leaderelection.go:194","timestamp":"2018-12-10T21:18:31.194806Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer persistentVolumeClaimInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.194908Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer fakeDataVolumeInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.194988Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer vmiInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195010Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer kubeVirtPodInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195028Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer configMapInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195059Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer vmInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195075Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer vmimInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195092Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer kubeVirtNodeInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195111Z"}
{"component":"virt-controller","level":"info","msg":"STARTING informer vmirsInformer","pos":"virtinformers.go:121","service":"http","timestamp":"2018-12-10T21:18:31.195128Z"}
{"component":"virt-controller","level":"info","msg":"Starting VirtualMachine controller.","pos":"vm.go:107","service":"http","timestamp":"2018-12-10T21:18:31.195201Z"}
{"component":"virt-controller","level":"info","msg":"Starting node controller.","pos":"node.go:109","service":"http","timestamp":"2018-12-10T21:18:31.195168Z"}
{"component":"virt-controller","level":"info","msg":"Starting VirtualMachineInstanceReplicaSet controller.","pos":"replicaset.go:106","service":"http","timestamp":"2018-12-10T21:18:31.195201Z"}
{"component":"virt-controller","level":"info","msg":"Starting migration controller.","pos":"migration.go:96","service":"http","timestamp":"2018-12-10T21:18:31.195278Z"}
{"component":"virt-controller","level":"info","msg":"Starting vmi controller.","pos":"vmi.go:167","service":"http","timestamp":"2018-12-10T21:18:31.195305Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:37.638008Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:47.639282Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:57.638635Z"}
{"component":"virt-controller","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:19:07.639120Z"}

$ kubectl logs virt-api-86c5774f57-xxz8t       
{"component":"virt-api","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.389520Z"}
{"component":"virt-api","level":"info","msg":"kubevirt-config ConfigMap does not exist. Using defaults.","pos":"feature-gates.go:66","timestamp":"2018-12-10T21:18:13.414878Z"}
{"component":"virt-api","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.414918Z"}
{"component":"virt-api","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.415782Z"}
{"component":"virt-api","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.421915Z"}
{"component":"virt-api","level":"warning","msg":"Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.","pos":"client_config.go:552","timestamp":"2018-12-10T21:18:13.428304Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 169.254.255.254:65134: EOF\n","pos":"server.go:2923","timestamp":"2018-12-10T21:18:18.204479Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 169.254.255.254:65240: EOF\n","pos":"server.go:2923","timestamp":"2018-12-10T21:18:28.204775Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 169.254.255.254:65422: EOF\n","pos":"server.go:2923","timestamp":"2018-12-10T21:18:38.204462Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 169.254.255.254:32842: EOF\n","pos":"server.go:2923","timestamp":"2018-12-10T21:18:48.204625Z"}
{"component":"virt-api","level":"info","msg":"http: TLS handshake error from 169.254.255.254:33022: EOF\n","pos":"server.go:2923","timestamp":"2018-12-10T21:18:58.204639Z"}
```

At this point, I can create VM resources, but no VMIs or associated Pods are ever created.

**What you expected to happen**:

KubeVirt components start up, successfully connect to each other, and VMs are scheduled

**How to reproduce it (as minimally and precisely as possible)**:

Not sure at this time. Happens consistently whenever I deploy KubeVirt.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`): v0.12.0-alpha.0
- Kubernetes version (use `kubectl version`): v0.10.11
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- Kernel (e.g. `uname -a`): 4.14.74
- Install tools:
- Others:

**Comment by fabiand** [comment]
Hm. Something is up in that area. It's not the first time we see something like this.

Can you try to reinstall kubevirt?

**Comment by fabiand** [comment]
Not that it is a fix, only to see if it helps

**Comment by terinjokes** [comment]
As stated above:

> Happens consistently whenever I deploy KubeVirt.

**Comment by terinjokes** [comment]
Some updates here:

* I forgot to start the VMs. I still get the errors about handshaking, so that might be entirely unrelated (maybe it's Prometheus being a bit eager to scrape?)
* I'm having an issue similar to #1128 (and maybe #1551) where I get errors saying that /dev/kvm is unavailable: `hardware emulation device '/dev/kvm' not present`:
  ```
  {"component":"virt-handler","kind":"","level":"error","msg":"Synchronizing the VirtualMachineInstance failed.","name":"testvm","namespace":"u-terin","pos":"vm.go:784","reason":"server error. command Launcher.Sync failed: \"hardware emulation device '/dev/kvm' not present\"","timestamp":"2018-12-10T23:26:17.037760Z","uid":"8579e526-fcd2-11e8-9c50-54ab3a16deea"}
  ```
* /dev/kvm is available on the worker, which is running natively.

**Comment by terinjokes** [comment]
Update: Removed Prometheus, still seeing these TLS handshake errors every 10 seconds.

**Comment by cynepco3hahue** [comment]
These errors exist also on the healthy environment, so the only fix that we need to add is to ignore this errors under the logging.

**Comment by rmohr** [comment]
I see where it is coming from. We use a `tcpProbe` as readiness check. That shoud be a httpGet with "HTTPS" scheme.

**Comment by cynepco3hahue** [comment]
@rmohr Sure will take a look.

**Comment by terinjokes** [comment]
Should I open another ticket (or update the one linked above) for the `/dev/kvm` problem? I'm still not sure what the issue is here.

**Comment by cynepco3hahue** [comment]
@terinjokes We already have `/dev/kvm` issue https://github.com/kubevirt/kubevirt/issues/1551, you can comment under it.

**Comment by fabiand** [comment]
@terinjokes what kind of host OS are you using, and did you configure emulation mode?

**Comment by terinjokes** [comment]
The /dev/kvm issue turned out to be a bug with another project's
MutatingWebhook not being Device Plugin aware. After patching it, the kvm
resource was being passed through, and the VM started correctly.

It took a bit of research to determine this project uses Device Plugin to
share /dev/kvm. Only after I knew was I able to find the blog posts.
